newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,172 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import HelpButton from '../../UI/HelpButton';
+import { Line, Column } from '../../UI/Grid';
+import ColorPicker from '../../UI/ColorField/ColorPicker';
+import MiniToolbar, { MiniToolbarText } from '../../UI/MiniToolbar';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+import { getSelectedEvents } from '../SelectionHandler';
+
+const gd = global.gd;
+
+const toolbarItemStyle = {
+  marginRight: 10,
+};
+
+const styles = {
+  sizeTextField: {
+    width: 90,
+    ...toolbarItemStyle,
+  },
+  toolbarItem: toolbarItemStyle,
+  checkbox: toolbarItemStyle,
+};
+
+type Props = {|
+  event: any,
+  open: boolean,
+  onClose: () => void,
+  onApply: () => void,
+|};
+
+type State = {|
+  searchText: string,
+  textValue: any,
+|};
+
+//TODO prendre en compte aussi les commentaires des groupes, vérifié le type de l'event getType()
+//Puis selon le cas retourné le text avec la methode approprié a l'event getComment() ou getName()
+//Voir comment recup l'event car ma façon de faire dans le state récupère ce text "\"
+export default class TextEditorDialog extends React.PureComponent<
+  Props,
+  State
+> {
+  recupereLecommentaire = () => {
+    return gd
+      .asCommentEvent(
+        getSelectedEvents(this.props.event).some(event => {
```

Comment:
Don't use `getSelectedEvents` inside TextEditorDialog, it's only to be used by the EventsSheet to manager its selection. Instead pass the event as a prop to TextEditorDialog

Reply:
If you read my other comments, once you've passed an `event` as a prop (a real event, not the selection!), then you can do something like:

```
const { event } = this.props;
const eventType = event.getType();
if (eventType === 'BuiltinCommonInstructions::Comment') {
  return gd.asCommentEvent(event).getComment();
} else if (eventType === 'BuiltinCommonInstructions::Group') {
  return gd.asGroupEvent(event).getName();
} else {
  console.error("Dialog was opened for an unsupported event type: " + eventType);
  return '';
}
```

Reply:
Fixed